docs/elasticsearch-specification.asciidoc

```diff
@@ -544,3 +545,28 @@ spec:
 ----
 
 Note that this will require restarting the pods.
+
+[id="{p}-prestop"]
+=== Pod PreStop hook
+
+When Elasticsearch `Pod` is terminated, its `Endpoint` is removed from the `Service` and the Elasticsearch process is terminated. As these two operations happen in parallel, a race condition exists. If the Elasticsearch process is already shut down, but the `Endpoint` is still a part of the `Service`, any new connection might fail. For more information, see link:https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods[Termination of pods].
+
+To address this issue and guarantee no downtime for new connections, ECK relies on PreStop lifecycle hook. This allows to delay Elasticsearch process termination until `Service` DNS record won't contain the IP of the `Pod`. First, PreStop lifecycle hook will keep querying DNS for `MAX_WAIT_SECONDS` (defaulting to 20) and then it will wait for `ADDITIONAL_WAIT_SECONDS` (defaulting to 1). Additional wait is needed to give clients time to use IP resolved just before DNS record was updated. The exact behavior is configurable using environment variables, for example:
```

Comment:
```suggestion
To address this issue and guarantee no downtime for new connections, ECK relies on a link:https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/[PreStop lifecycle hook]. This waits to terminate the Elasticsearch process until the `Service` DNS record does not contain the IP of the `Pod`. First, the PreStop lifecycle hook will keep querying DNS for `MAX_WAIT_SECONDS` (defaulting to 20) and then it will wait for `ADDITIONAL_WAIT_SECONDS` (defaulting to 1). Additional wait is used to give clients time to use an IP resolved just before DNS record was updated. The exact behavior is configurable using environment variables, for example:
```